### PR TITLE
DataLinks: Removes getDataSourceSettingsByUid from applyFieldOverrides 

### PR DESCRIPTION
**Pull request by torkelo**
https://github.com/grafana/grafana/pull/25613 added getDataSourceSettingsByUid as a temporary measure that we were to remove in follow-up PR. 

Was about to start doing some work on DataSourceSrv to make the get function work on name & uid transparently and discovered that all getDataSourceSettingsByUid was used for was to just lookup the name. So why not get the name when we build the link model instead? And that looked easy, gave it a shot. Tested with loki + jaeger.  

Todo:
* [x] Update unit tests (could need help with this) 
* [ ] Update X-ray datasource

So I added datasourceName to internal link type, but think we can get rid of that soon as we well. Just need to make data source lookup possible with name or uid, so then getInternalHref can use uid for data source instead of name. And we can just set title instead (in data source code) when the internal link model is built. 

Related to #29417

After #29449 is merged we should be able to remove datasourceName and just set title on the link as normal, as with #29449 we should be able to target the data source with the uid instead of the name

### CODE REVIEW

**Comment by aocenas** [comment]
Nice yeah did not realize this before but it makes sense. Maybe we could/should switch Explore links to use UIDs instead of names?

File: public/app/plugins/datasource/loki/result_transformer.ts [hunk 1]
```diff
@@ -387,9 +387,13 @@ export const enhanceDataFrame = (dataFrame: DataFrame, config: LokiOptions | nul
  * Transform derivedField config into dataframe field with config that contains link.
  */
 function fieldFromDerivedFieldConfig(derivedFieldConfigs: DerivedFieldConfig[]): Field<any, ArrayVector> {
+  const dataSourceSrv = getDataSourceSrv();
+
   const dataLinks = derivedFieldConfigs.reduce((acc, derivedFieldConfig) => {
     // Having field.datasourceUid means it is an internal link.
     if (derivedFieldConfig.datasourceUid) {
+      const dsSettings = dataSourceSrv.getDataSourceSettingsByUid(derivedFieldConfig.datasourceUid);
```
Comment (dprokop) [inline public/app/plugins/datasource/loki/result_transformer.ts:395]: Guess this won't be necessary at some point?

**Review by dprokop — APPROVED** [review]
LGTM!

File: public/app/plugins/datasource/loki/result_transformer.ts (hunk 1 above)
Comment (torkelo) [reply public/app/plugins/datasource/loki/result_transformer.ts:395]: I think we will likely always have to lookup the data source to set a link title at least. Or we let the user set the link title when they define the derived field

Comment (dprokop) [reply public/app/plugins/datasource/loki/result_transformer.ts:395]: OK I see thanks!

**Comment by torkelo** [comment]
> Nice yeah did not realize this before but it makes sense. Maybe we could/should switch Explore links to use UIDs instead of names?

Yes, will try to support that in https://github.com/grafana/grafana/pull/29449 or in follow-up PR to that one. But we still need to set the link name somewhere, either in the data source result transformer or when the derived field is configured

**Comment by torkelo** [comment]
Merging this, created an issue to followup with x-ray fix: https://github.com/grafana/grafana/issues/29506 

But want to wait with that, because depending on if we can get explore to only use data source uid, we can set the title and skip this datasourceName fully

**Pull request merged**